velox/docs/functions/spark/decimal.rst

```diff
@@ -2,47 +2,92 @@
 Decimal Operators
 =================
 
-When calculating the result precision and scale of arithmetic operators,
-the formulas follow Hive which is based on the SQL standard and MS SQL:
+The result precision and scale computation of arithmetic operators contains two stages.
+In the first stage, the formulas follow Hive which is based on the SQL standard and MS SQL.
```

Comment:
perhaps,

First stage computes precision and scale using formulas based on the SQL standard, MS SQL and Hive. The result may exceed maximum allowed precision of 38.

Second stage caps precision at 38 and either reduces the scale or not depending on allow-precision-loss flag.

For example, addition of decimal(38, 7) and decimal(10, 0) requires precision of 39 and scale of 7. Since precision exceeds 38 it needs to be capped. When allow-precision-loss, precision is capped at 38 and scale is reduced by 1 to 6. When allow-precision-loss, precision is capped at 38 as well, but scale is kept at 7. With allow-precision-loss all additions will succeed, but accuracy (number of digits after period) of some operations will be reduced. Without allow-precision-loss, some additions will fail.

For example,

The following addition fails when allow-precision-loss is false:

select cast('9999999999999999999999999999999.2345678' as decimal(38, 7)) + cast('1' as decimal(10, 0))

... TBD: include an error

and succeeds with allow-precision-loss is true:

... TBD: include result and highlight precision loss